src/GraphNavigator/DeserializationGraphNavigator.php

```diff
@@ -107,9 +108,24 @@ public function accept($data, ?array $type = null)
         if (null === $type) {
             throw new RuntimeException('The type must be given for all properties when deserializing.');
         }
+
+        $shouldDeserializeNull = false;
+        $metadataStack = $this->context->getMetadataStack();
+
+        if (!$metadataStack->isEmpty()) {
```

Comment:
Really not convinced by this `if`. Both for performance and clarity. But I undersatdn that this lines are the "core" of how the whole feature works... Am I right?

Reply:
If you could break compatibility, what would you do?

Reply:
I don't like it either but the problem was that we need the `deserializeNull` option in both GraphNavigator and Visitor, and this is the only way we can get the current property that is being deserialized and its option.

>If you could break compatibility, what would you do?

Can answer this in a few days when I investigate it better